### PR TITLE
CRM-19770 - Add is_star column to civicrm_activity

### PR DESCRIPTION
**Pull request by colemanw**
* [CRM-19770: Favorite \(starred\) activities](https://issues.civicrm.org/jira/browse/CRM-19770)

### CODE REVIEW

File: CRM/Upgrade/Incremental/php/FourSeven.php [hunk 1]
```diff
@@ -280,14 +280,8 @@ public function upgrade_4_7_14($rev) {
    * @param string $rev
    */
   public function upgrade_4_7_15($rev) {
-    $this->addTask('CRM-19723 - Add icon column to civicrm_option_value', 'addColumn',
-      'civicrm_option_value', 'icon', "varchar(255) COMMENT 'crm-i icon class' DEFAULT NULL");
     $this->addTask('CRM-19626 - Add min_amount column to civicrm_price_set', 'addColumn',
```
Comment (totten) [inline CRM/Upgrade/Incremental/php/FourSeven.php]: For posterity: We discussed this one on Mattermost. This step is in an odd position where the `4.7.15` uses a SQL-backed upgrade and `4.7.16` uses a newer PHP-backed upgrade. However, testing shows that they end up in the same place, and they still appear mutually compatible, and the `4.7.16` code is better.

File: CRM/Upgrade/Incremental/php/FourSeven.php [hunk 2]
```diff
@@ -297,10 +291,16 @@ public function upgrade_4_7_15($rev) {
    * @param string $rev
    */
   public function upgrade_4_7_16($rev) {
-    $this->addTask(ts('Upgrade DB to %1: SQL', array(1 => $rev)), 'runSql', $rev);
+    $this->addTask('CRM-19723 - Add icon column to civicrm_option_value', 'addColumn',
+      'civicrm_option_value', 'icon', "varchar(255) COMMENT 'crm-i icon class' DEFAULT NULL");
+    $this->addTask('CRM-19769 - Add color column to civicrm_tag', 'addColumn',
+      'civicrm_tag', 'color', "varchar(255) COMMENT 'Hex color value e.g. #ffffff' DEFAULT NULL");
```
Comment (totten) [inline CRM/Upgrade/Incremental/php/FourSeven.php]: Belated nitpick... and this isn't going to block #9629... but `varchar(255)` seems much longer than a color needs. `varchar(8)` is probably enough, and `varchar(32)` would also be pretty comfortable.

**Comment by totten** [comment]
Confirmed that the revised upgrade works -- for both the new field and the refactored upgrade steps.

File: CRM/Upgrade/Incremental/php/FourSeven.php (hunk 2 above)
Comment (colemanw) [inline CRM/Upgrade/Incremental/php/FourSeven.php]: @totten I actually obsessed about that a little before looking it up and discovering (lo and behold)... it doesn't matter!
http://stackoverflow.com/questions/8295131/best-practices-for-sql-varchar-column-length

Comment (totten) [inline CRM/Upgrade/Incremental/php/FourSeven.php]: @colemanw interesting, my mental model of `varchar(nnn)` was a bit wrong!

FWIW... one of those answers cites http://stackoverflow.com/a/1151833/4195300, but it seems to give a more complicated view -- i.e. the length shouldn't affect disk usage but may affect query performance.

**Pull request merged**